utils.py

```diff
@@ -96,104 +134,54 @@ def sendEMail(dests, name, skel, extraFiles=[], cc=None, bcc=None, replyTo=None,
-	"""
-	Internal function for using Google App Engine Email processing API.
-	"""
-	return
-	headers, data = conf["viur.emailRenderer"](skel, name, dests, **kwargs)
-
-	xheader = {}
-
-	if "references" in headers:
-		xheader["References"] = headers["references"]
-
-	if "in-reply-to" in headers:
-		xheader["In-Reply-To"] = headers["in-reply-to"]
-
-	if xheader:
-		message = mail.EmailMessage(headers=xheader)
-	else:
-		message = mail.EmailMessage()
-
-	mailfrom = "viur@%s.appspotmail.com" % projectID
-
-	if "subject" in headers:
-		message.subject = "=?utf-8?B?%s?=" % base64.b64encode(headers["subject"].encode("UTF-8"))
-	else:
-		message.subject = "No Subject"
-
-	if "from" in headers:
-		mailfrom = headers["from"]
-
 	if conf["viur.emailSenderOverride"]:
-		mailfrom = conf["viur.emailSenderOverride"]
-
-	if isinstance(dests, list):
-		message.to = ", ".join(dests)
-	else:
-		message.to = dests
-
-	if cc:
-		if isinstance(cc, list):
-			message.cc = ", ".join(cc)
-		else:
-			message.cc = cc
+		sender = conf["viur.emailSenderOverride"]
+	elif sender is None:
+		sender = f"viur@{projectID}.appspotmail.com"
 
-	if bcc:
-		if isinstance(bcc, list):
-			message.bcc = ", ".join(bcc)
-		else:
-			message.bcc = bcc
+	handler = conf.get("viur.emailHandler")
 
-	if replyTo:
-		message.reply_to = replyTo
+	if handler is None:
+		raise errors.InvalidConfigException("No emailHandler specified!")
+	elif not callable(handler):
+		raise errors.InvalidConfigException("Invalid emailHandler configured, no email will be sent!")
 
-	message.sender = mailfrom
-	message.html = data.replace("\x00", "").encode('ascii', 'xmlcharrefreplace')
+	body = conf["viur.emailRenderer"](dests, file, template, skel, template_params, **kwargs)
 
-	if len(extraFiles) > 0:
-		message.attachments = extraFiles
-	message.send()
-	return True
+	return handler(dests=dests, sender=sender, cc=cc, bcc=bcc, replyTo=replyTo,
+				   subject=subject, header=header, body=body, attachments=attachments, *args, **kwargs)
 
 
-def sendEMailToAdmins(subject, body, sender=None):
+def sendEMailToAdmins(subject: str, body: str, sender: str = None):
 	"""
-		Sends an e-mail to the appengine administration of the current app.
-		(all users having access to the applications dashboard)
+		Sends an e-mail to the root users of the current app.
 
 		:param subject: Defines the subject of the message.
-		:type subject: str
 
 		:param body: Defines the message body.
-		:type body: str
 
 		:param sender: (optional) specify a different sender
-		:type sender: str
 	"""
-	return
 	if not sender:
-		sender = "viur@%s.appspotmail.com" % projectID
+		sender = f"viur@{projectID}.appspotmail.com"
+
+	if "user" in dir(conf["viur.mainApp"]):
+		users = []
+		for userSkel in conf["viur.mainApp"].user.viewSkel().all().filter("access =", "root").fetch():
+			users.append(userSkel["name"])
+
+		if users:
+			return sendEMail(dests=users, subject=subject, template=body, sender=sender)
```

Comment:
should this run outside a try-except block, especially with the critical loggings below?

Reply:
Hmm. Good point. I think _sendEMailToAdmins_ should (re-)raise an error if the sending failed, so you're able to handle this in your application. But of course it might be a good behavior to log the errors as well.